packages/storage/src/index.ts

```diff
@@ -394,7 +394,7 @@ function createStorage(): WxtStorage {
               (await migrations?.[migrateToVersion]?.(migratedValue)) ??
               migratedValue;
           } catch (err) {
-            throw Error(`v${migrateToVersion} migration failed for "${key}"`, {
+            throw new MigrationError(key, migrateToVersion, {
```

Comment:
Vitest 3 introduced more strict error equality checks, causing the test to fail since we weren't using the actual migration error class. Basically, the newer vitest caught a bug